provider/core/src/export/mod.rs

```diff
@@ -176,11 +185,48 @@ impl DataExporter for MultiExporter {
             .try_for_each(|e| e.flush_singleton(marker, payload))
     }
 
-    fn flush(&self, marker: DataMarkerInfo) -> Result<(), DataError> {
-        self.0.iter().try_for_each(|e| e.flush(marker))
+    fn flush(
+        &self,
+        marker: DataMarkerInfo,
+        deduplication: DeduplicationStrategy,
+    ) -> Result<(), DataError> {
+        self.0
+            .iter()
+            .try_for_each(|e| e.flush(marker, deduplication))
     }
 
     fn close(&mut self) -> Result<(), DataError> {
         self.0.iter_mut().try_for_each(|e| e.close())
     }
 }
+
+/// Choices for determining the deduplication of locales for exported data payloads.
+///
+/// Deduplication affects the lookup table from locales to data payloads. If a child locale
+/// points to the same payload as its parent locale, then the child locale can be removed from
+/// the lookup table. Therefore, all deduplication strategies guarantee that data requests for
+/// selected locales will succeed so long as fallback is enabled at runtime (either internally
+/// or externally). They also do not impact which _payloads_ are included: only the lookup table.
+///
+/// Comparison of the deduplication strategies:
+///
+/// | Name | Data file size | Supported locale queries? | Needs runtime fallback? |
+/// |---|---|---|---|
+/// | [`Maximal`] | Smallest | No | Yes |
+/// | [`RetainBaseLanguages`] | Small | Yes | Yes |
+/// | [`None`] | Medium/Small | Yes | No |
+///
+/// [`Maximal`]: DeduplicationStrategy::Maximal
+/// [`RetainBaseLanguages`]: DeduplicationStrategy::RetainBaseLanguages
+/// [`None`]: DeduplicationStrategy::None
+#[non_exhaustive]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum DeduplicationStrategy {
+    /// Removes from the lookup table any locale whose parent maps to the same data.
+    Maximal,
+    /// Removes from the lookup table any locale whose parent maps to the same data, except if
+    /// the parent is `und`.
+    RetainBaseLanguages,
+    /// Keeps all selected locales in the lookup table.
+    None,
+}
```

Comment:
(splitting this specific discussion to a different thread)

DeduplicationStrategy is very much an artifact of locale fallbacking and datagen. It is not something that the end user / client should care about: only the developer configuring their ICU4X deployment. The only thing that impacts the end user / client is whether they can run "supported locale" queries. The fact that such queries work on RetainBaseLanguages and None is a low-level detail. The DataExporter trait seems like the right place to define client-oriented enums instead of sysops-oriented enums.

Reply:
> The fact that such queries work on RetainBaseLanguages and None is a low-level detail

But it is a detail that the exporters can be exposed to. `DataExporter` is not client-facing, and if a client wants to implement this, `DeduplicationStrategy` is not something that should be abstracted away.

Reply:
> DataExporter is not client-facing

I see DataExporter as the first client-facing trait in the pipeline. A DataProvider is directly derived from a DataExporter, and it should share the same types. It should not be up to a DataExporter to decide how to map datagen options to client options; that decision should have already been made before the data reaches the DataExporter.

We can see this by how you need to move DeduplicationStrategy into `icu_provider` where it has no business being. (I realize that an earlier draft PR of mine did this, too, but (1) it was only a draft and (2) after I did it, I wanted to undo it)